cpp/include/cudf/detail/datetime_ops.cuh

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cuda/std/chrono>
+
+namespace cudf {
+namespace datetime {
+namespace detail {
+
+template <typename Timestamp, typename MonthType>
+__device__ Timestamp add_calendrical_months_with_scale_back(Timestamp time_val,
+                                                            MonthType months_val)
+{
+  using namespace cuda::std::chrono;
+
+  // Get the days component from the input
+  auto const days_since_epoch = floor<days>(time_val);
+
+  // Add the number of months
+  year_month_day ymd{days_since_epoch};
+  ymd += duration<int32_t, months::period>{months_val};
```

Comment:
Why not just use `months` directly? 
```suggestion
  ymd += std::chrono::months{months_val};
```

Reply:
And thinking further on this, it seems like the `months_val` parameter type should be just changed to `cuda::std::chrono::months`.

Reply:
To do this we also need to construct `months` outside of the function because `duration` ctor with rep value is `explicit`: https://en.cppreference.com/w/cpp/chrono/duration/duration